### PR TITLE
refactor(druid): port to sqlglot

### PR DESCRIPTION
**Pull request by cpcloud**
This PR ports the Druid backend to sqlglot

### CODE REVIEW

File: ibis/backends/druid/__init__.py [hunk 1]
```diff
@@ -107,33 +110,74 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
...
+
+        try:
+            df = pd.DataFrame.from_records(
+                cursor, columns=schema.names, coerce_float=True
             )
-            return super()._get_sqla_table(name, autoload=autoload, **kwargs)
+        except Exception:
+            # clean up the cursor if we fail to create the DataFrame
+            #
+            # in the sqlite case failing to close the cursor results in
```
Comment (kszucs) [inline ibis/backends/druid/__init__.py:135]: sqlite case?

File: ibis/backends/druid/__init__.py [hunk 2]
```diff
@@ -107,33 +110,74 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
...
+            # artificially locked tables
+            cursor.close()
+            raise
+        df = PandasData.convert_table(df, schema)
+        return df
+
+    def create_table(
```
Comment (kszucs) [inline ibis/backends/druid/__init__.py:142]: Wasn't this supported before?

File: ibis/backends/druid/compiler.py [hunk 3]
```diff
@@ -1,33 +1,232 @@
...
+from ibis.backends.base.sqlglot.compiler import NULL, SQLGlotCompiler
+from ibis.backends.base.sqlglot.datatypes import DruidType
 from ibis.expr.rewrites import rewrite_sample
 
 
-class DruidExprTranslator(AlchemyExprTranslator):
-    _registry = operation_registry.copy()
-    _rewrites = AlchemyExprTranslator._rewrites.copy()
-    _dialect_name = "druid"
+# Is postgres the best dialect to inherit from?
```
Comment (kszucs) [inline ibis/backends/druid/compiler.py:20]: Do you want to keep this comment? :)

File: ibis/backends/druid/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/druid/__init__.py:135]: yeah, copypasta. ideally we can clean this up a bit later

Comment (cpcloud) [reply ibis/backends/druid/__init__.py:135]: though not urgent by any means

File: ibis/backends/druid/__init__.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/druid/__init__.py:142]: I don't think so. Lemme check

Comment (cpcloud) [reply ibis/backends/druid/__init__.py:142]: Druid doesn't have a `CREATE TABLE` syntax

File: ibis/backends/druid/compiler.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/druid/compiler.py:20]: I mean, it's still an open question

File: ibis/backends/druid/tests/conftest.py [hunk 4]
```diff
@@ -101,7 +103,14 @@ class TestConf(ServiceBackendTest):
...
     service_name = "druid-middlemanager"
-    deps = ("pydruid.db.sqlalchemy",)
+    deps = ("pydruid.db",)
+
+    @property
+    def functional_alltypes(self) -> ir.Table:
+        t = self.connection.table(
+            self.default_identifier_case_fn("functional_alltypes")
+        )
+        return t.mutate(bool_col=1 - t.id % 2)
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ibis/backends/druid/tests/conftest.py:113]: Can you add this as a comment?

**Review by kszucs — APPROVED** [review]
LGTM

File: ibis/backends/druid/tests/conftest.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/druid/tests/conftest.py:113]: I verified that it is also broken against `main`.

File: ibis/backends/druid/compiler.py (hunk 3 above)
Comment (kszucs) [reply ibis/backends/druid/compiler.py:20]: Apparently it is working, so I'd remove the comment.

File: ibis/backends/druid/__init__.py (hunk 2 above)
Comment (kszucs) [reply ibis/backends/druid/__init__.py:142]: Sweet, resolved then.

File: ibis/backends/druid/tests/conftest.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/druid/tests/conftest.py:113]: Yep, adding as a comment.

[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/backends/druid/tests/conftest.py:115]: Done.

**Pull request merged**